hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.tsx

```diff
@@ -184,56 +212,89 @@ const MainSection = ({ name, total, quantityAvailable, categories, img }) => {
+                                className={styles.categoryItem}
+                                key={i}
+                            />
+                        ))}
                 </div>
             </div>
-            <img src={img} alt={name} />
+            <img src={picture} alt={name} />
         </div>
     );
 };
 
-export const ProductOverview = ({ detail, addToCart, isVisible, handleClose }) => (
-    <SideSheetRight
-        title="Product Overview"
-        isVisible={isVisible}
-        handleClose={handleClose}
-    >
-        <div className={styles.productOverview}>
-            <div className={styles.productOverviewDiv}>
-                <MainSection
-                    type={detail.type}
-                    name={detail.name}
-                    total={detail.total}
-                    quantityAvailable={detail.quantityAvailable}
-                    categories={detail.category}
-                    img={detail.img}
-                />
-                <DetailInfoSection
-                    manufacturer={detail.manufacturer}
-                    model_num={detail.model_num}
-                    datasheet={detail.datasheet}
-                    notes={detail.notes}
-                    constraints={detail.constraints}
-                />
-            </div>
+export const ProductOverview = ({
+    showAddToCartButton,
+}: {
+    showAddToCartButton: boolean;
+}) => {
+    const dispatch = useDispatch();
 
-            {addToCart && (
-                <EnhancedAddToCartForm
-                    handleSubmit={addToCart}
-                    requestFailure={false}
-                    quantityAvailable={detail.quantityAvailable}
-                    constraintMax={detail.constraintMax}
-                />
+    const isProductOverviewVisible: boolean = useSelector(
+        isProductOverviewVisibleSelector
+    );
+    const hardware: Hardware | null = useSelector(hardwareBeingViewedSelector);
+    const allCategories = useSelector(categorySelectors.selectEntities);
+
+    const categories: (Category | undefined)[] = hardware
+        ? hardware.categories.map((categoryId) => allCategories[categoryId])
+        : [];
+
+    const constraints: (string | undefined)[] =
+        categories.length > 0
+            ? [`Max ${hardware?.max_per_team} of this item`].concat(
+                  categories.map(
+                      (category) =>
+                          `Max ${category?.max_per_team} of items under category ${category?.name}`
+                  )
+              )
+            : [];
+
+    const categoryNames: (string | undefined)[] =
+        categories.length > 0 ? categories.map((category) => category?.name) : [];
+
+    const closeProductOverview = () => dispatch(removeProductOverviewItem());
+
+    return (
+        <SideSheetRight
+            title="Product Overview"
+            isVisible={isProductOverviewVisible}
+            handleClose={closeProductOverview}
+        >
+            {hardware && (
```

Comment:
I don't see how this modal would be rendered with hardware that doesn't exist because it's accessed from the inventory page. Fetching more hardware makes sense but I don't see the case where that would be needed

Reply:
also if hardware is null, how do we attempt to fetch hardware? we don't know what hardware we want

Reply:
Oh good point, this isn't like the cart cards where we have a hardware ID to fetch. 

As long as we make sure that the hardware is in the store before dispatching an action to load it, we should be fine. The edge case I was considering is that we only get hardware IDs back from the order API, so we just need to make sure all of those hardware are loaded before rendering anywhere (which is probably required anyway).